### ISSUE TITLE
RNDIS USB networking issue on iPad Pro M4 and RPi 5, but works on iPad Pro M2 and RPi 5

### ISSUE THREAD

**Issue by jun3010me** [verxion/RaspberryPi#1]
## Summary

The RNDIS USB network driver fails to load and work properly on the newer iPad Pro M4 chip model and Raspberry Pi 5, while it works as expected on the Raspberry Pi 4.

## Environment

- Raspberry Pi 4 Model B Rev 1.5 (Updated to latest Raspberry Pi OS release)
- Raspberry Pi 5 Model B (Updated to latest Raspberry Pi OS release)  
- iPad Pro M4 (2024) running iPadOS 17.x
- iPad Pro M2 (2022) running iPadOS 17.x

## Behavior

On the Raspberry Pi 5, the RNDIS driver loads successfully, and the iPad Pro M2 can connect to it via USB and use it as an Ethernet adapter.

However, on the Raspberry Pi 5 and the newer iPad Pro M4, the following issues occur:

1. On Raspberry Pi 5, the RNDIS driver fails to load successfully during boot.
2. On iPad Pro M4, even when the Raspberry Pi RNDIS driver seems to load, the "Ethernet" option does not show up in the iPad's Settings app to enable the USB-Ethernet connection.

## Steps to Reproduce

1. Connect the Raspberry Pi to the iPad Pro via USB cable
2. On Raspberry Pi, check `dmesg` output for RNDIS driver loading
3. On iPad Pro, go to Settings -> Ethernet and check if the USB-Ethernet adapter is listed

Any insights into why this regression happens on the newer Raspberry Pi 5 and iPad Pro M4 chip would be appreciated. A solution to properly load the RNDIS driver and get USB-Ethernet working on these devices is required.

**Comment by JDBarndt** [comment]
Any luck on this? I believe I'm running into a similar situation with a Pi 5 and the new M4 iPad Pro.

**Comment by OsvaldoEst** [comment]
Any help?

**Comment by qeleb** [comment]
+1
Followed all of the steps perfectly. No Ethernet connections show up on my M4 iPad Pro
…even though it shows up on my iphone and other devices 😬

**Comment by jun3010me** [comment]
I'll post it after a long time. I tried various methods and wrote what was effective on my blog. Thank you very much for your comment.

https://jun3010.me/raspberry-pi5-usbotg-gether.html

**Comment by verxion** [comment]
Hey everyone, have had some family issues of late, sorry for not following up, had to set priorities elsewhere for a while. 

There is no need to change any of the OS/Software/Config process here for M4 iPad Pro. There appears to be a hardware issue with it though. Instead of using a straight USB-C to USB-C cable, you just have to go USB-C -> USB-A female -> USB-A male -> USB-C with two cable, two dongles or a cable and a dongle. 

As just an example these two will work:

CableCreation 6 inch USB C Cable... https://www.amazon.com/dp/B01CZVEUIE

UGREEN USB C to USB A Adapter 2... https://www.amazon.com/dp/B0C4KWHRT5

I’m NOT saying to get these specific ones, just passing on that these definitely work.

**Comment by jclark** [comment]
See https://github.com/raspberrypi/linux/issues/6569 

TLDR use `rpi-eeprom-config -e` to add `PSU_MAX_CURRENT=3000`

If you go through a USB A dongle, then the Pi 5 won't get the full 5V3A of power.
